### DISCUSSION TITLE
Setup: energy dashboard Return to grid

### DISCUSSION THREAD

**Discussion by peted-davis** [BottlecapDave/HomeAssistant-OctopusEnergy#710] [Q&A] [answered]
Apologies if I have missed something obvious.

I have an Octopus Home Mini and I want to setup the energy dashboard to use this data. I have followed the setup guide and have set the "Grid consumption" field to the Current Accumulative Consumption entity. We have solar panels on the house and I also want to monitor the amount we return to the grid. Is there an entity available that allows me to track the amount we export? I can see that the Current Demand entity is negative when we are exporting, but the Current Accumulative Consumption entity doesn't go below zero. 

We don't have an export meter

**Comment by Hexadecimalism** [comment]
I suspect the situation is still as listed here https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/395#discussioncomment-6891684 which is unfortunate. Much like you, I'd love to have a real time export. I'm on Solis and unfortunately, Solis doesn't seem to have real time export (that I can find yet).

**Reply by peted-davis** [reply]
Yes, that sounds like the case. 

As the Home Mini shows negative values when exporting, I have made a workaround that integrates the negative usage to give the total export in kWh. This is only a rough value though as the sensor only updates once per minute. Ideally it needs to update more often.

**Reply by BottlecapDave** [reply, marked as answer]
Hello. I'm afraid this information is correct and the Octopus Home Mini still doesn't expose export information at a consumption level but only in the live view (see "does the home mini support export" in https://octopus.energy/octopus-home-mini-faq/#ohmfaqs). 

Because of rate limiting on the home mini APIs and the stability of the APIs, the integration can't retrieve this data any more than once per minute. This frequency as pointed out is not frequent enough to build a sensor with an accurate value.

Hopefully the upcoming Octopus Home Pro can provide this data in the future.

**Reply by peted-davis** [reply]
Thanks for confirming this. I haven't heard of the Octopus Home Pro, but will keep an eye out for it!
